### ISSUE TITLE
Implement support for cloning up to a recovery target specified by name

### ISSUE THREAD

**Issue by alexeyklyukin** [zalando/postgres-operator#335]
In all supported versions of PostgreSQL it is possible to specify the `recovery_target_name` option in `recovery.conf` to stop recovery when the named recovery point is reached. Such a point can be created with the call to `pg_create_restore_point()` on the master. Named recovery points are easier than precise timestamps to remember and interpret by humans .

Implementing support for named recovery points in the operator is easy, but constitutes only half of the deal: the Spilo docker image should be modified as well. Since WAL-E is currently called to fetch the specific base backup based on the timestamp passed to it, we need to figure out how to work around this timestamp requirement.

One idea is to take the assumption that the named recovery point always belongs to the latest basebackup; the other one is to specify both the recovery point and the approximate timestamp. Yet another one is to always start from the earliest available base backup, although that could make the recovery process significantly longer than necessary.

**Comment by mgomezch** [comment]
Sounds like WAL-e could use a feature here to allow specifying a named
recovery point in backup-push? If it were included in the backup metadata,
it could be selected easily. I guess it would have to be taken right after
taking the backup, though.

**Comment by alexeyklyukin** [comment]
Right, but WAL-E has no way of figuring this out. I assume the label is stored internally in the WAL file and doesn't affect any visible properties of the file, such as its name.

**Comment by FxKu** [comment]
I think, this idea has no priority atm and in the near future.
